### PR TITLE
Bump python dependencies & add initialize capability

### PR DESCRIPTION
**Pull request by dcermak**

### CODE REVIEW

File: salt_lsp/server.py [hunk 1]
```diff
@@ -158,6 +161,10 @@ def setup_salt_server_capabilities(server: SaltServer) -> None:
     the provided server.
     """
 
+    @server.feature(INITIALIZE)
+    def initialze(initialization_params: InitializeParams) -> InitializeResult:
```
Comment (agraul) [inline salt_lsp/server.py:165]: Typo:
```suggestion
    def initialize(initialization_params: InitializeParams) -> InitializeResult:
```

**Review by agraul — COMMENTED** [review]
I don't quite understand why it's needed. I expected that it is enough to have `super().__init__(protocol_cls=SaltLspProto)` in `SaltServer()` in order to get the overridden `lsp_initialize` called :confused:

**Comment by dcermak** [comment]
> I don't quite understand why it's needed. I expected that it is enough to have `super().__init__(protocol_cls=SaltLspProto)` in `SaltServer()` in order to get the overridden `lsp_initialize` called confused

I am not sure why it is necessary either, but if you omit the function call, then the test lsp server will not launch for the unit tests :shrug:

**Comment by agraul** [comment]
> > I don't quite understand why it's needed. I expected that it is enough to have `super().__init__(protocol_cls=SaltLspProto)` in `SaltServer()` in order to get the overridden `lsp_initialize` called confused
> 
> I am not sure why it is necessary either, but if you omit the function call, then the test lsp server will not launch for the unit tests shrug

I saw that as well, but something else is fishy. After checking out this branch, the initialization happens again and again

**Comment by agraul** [comment]
I think I understand what happens, more or less.

We register our "user" `INITIALIZE` feature in `salt_lsp.server`. This feature calls `salt_lsp.workspace.SaltLspProto.lsp_initialize`. That initialization routine, that's only used to replace its `self.workspace` with an `SlsFileWorkspace` instance, calls `super().lsp_initialize`. `super().lsp_initialize` is resolved to `pygls.protocol.LanguageServerProtocol.lsp_initialize`, which is decorated with `lsp_method(INITIALIZE)`. Once that function is called, it goes through the `call_user_func` wrapper which calls the built-in feature and then the user-supplied feature. In other words, we loop around.

I'm not 100% sure on all details, e.g. why only the "user" feature is called in the first place.

**Comment by agraul** [comment]
Looks like I need to add a test to get a positive test coverage delta. Other than that, I think this PR is in a decent state.

**Comment by dcermak** [comment]
@agraul I don't want to block this PR due to a coverage drop, so if you're happy with the current state, I'd merge it anyway. WDYT?

**Comment by agraul** [comment]
I think it's good enough to merge as it is  :+1:

**Pull request merged**